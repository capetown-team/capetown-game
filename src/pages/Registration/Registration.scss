@import '../../styles/common/login';

.form__title {
    margin-top: 48px;
}

.form__row {
    margin-top: 12px;
}

.form__row-button {
    margin-top: 12px;
    margin-bottom: 30px;
    text-align: center;
    display: flex;
    justify-content: center;
}
