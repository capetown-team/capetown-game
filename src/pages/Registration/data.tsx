import { FormDataType } from '@/types.d';

export const data: FormDataType = {
  email: {
    element: 'input',
    value: '',
    config: {
      name: 'email_input',
      type: 'email',
      placeholder: 'Введите свой email',
      label: 'Почта'
    },
    validation: {
      required: true,
      email: true
    },
    valid: false,
    touched: false,
    validationMessage: ''
  },
  login: {
    element: 'input',
    value: '',
    config: {
      name: 'login_input',
      type: 'text',
      placeholder: 'Введите свой логин',
      label: 'Логин'
    },
    validation: {
      required: true,
      minLength: 4
    },
    valid: false,
    touched: false,
    validationMessage: ''
  },
  first_name: {
    element: 'input',
    value: '',
    config: {
      name: 'first_name_input',
      type: 'text',
      placeholder: 'Введите ваше имя',
      label: 'Имя'
    },
    validation: {
      required: true,
      minLength: 2
    },
    valid: false,
    touched: false,
    validationMessage: ''
  },
  password: {
    element: 'input',
    value: '',
    config: {
      name: 'password_input',
      type: 'password',
      placeholder: 'Введите пароль',
      label: 'Пароль'
    },
    validation: {
      required: true,
      minLength: 8,
      password: true
    },
    valid: false,
    touched: false,
    validationMessage: ''
  },
  confirmPassword: {
    element: 'input',
    value: '',
    config: {
      name: 'confirmPassword_input',
      type: 'password',
      placeholder: 'Введите новый пароль (ещё раз)',
      label: 'Пароль (ещё раз)'
    },
    validation: {
      required: true,
      confirm: 'password'
    },
    valid: false,
    touched: false,
    validationMessage: ''
  }
};
