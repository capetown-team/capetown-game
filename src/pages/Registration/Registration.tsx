import React from 'react';

import './Registration.scss';
import { Button } from '../../components/Button';
import { Input } from '../../components/Input';

export const Registration = () => (
    <div className="body">
        <form className="login" >
            <div className="title">
                Регистрация
            </div>
            <div className="row">
                <div className="title-input"> Email </div>
                <Input id="email" placeHolder="Email"></Input>
            </div>
            <div className="row">
                <div className="title-input"> Логин </div>
                <Input id="lobin" placeHolder="Логин"></Input>
            </div>
            <div className="row">
                <div className="title-input"> Имя </div>
                <Input id="name" placeHolder="Имя"></Input>
            </div>
            <div className="row">
                <div className="title-input"> Пароль </div>
                <Input id="password" placeHolder="Пароль"></Input>
            </div>
            <div className="row">
                <div className="title-input"> Пароль (еще раз) </div>
                <Input id="passwordConfirm" placeHolder="Пароль (еще раз) "></Input>
            </div>
            <div className="row-button">
                <Button>Зарегистрироваться</Button>
            </div>

        </form>
    </div>
);