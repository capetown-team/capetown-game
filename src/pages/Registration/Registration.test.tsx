import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { Registration } from './Registration';

describe('<Registration />', () => {
  it('should renders correct <Registration />', () => {
    const wrapper = shallow(<Registration />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
