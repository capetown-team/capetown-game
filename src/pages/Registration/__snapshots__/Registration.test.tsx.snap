// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Registration /> should renders correct <Registration /> 1`] = `
<div
  className="form__wrapper"
>
  <form
    className="form__login"
  >
    <div
      className="form__title"
    >
      Регистрация
    </div>
    <div
      className="form__row"
    >
      <div
        className="form__title-input"
      >
        Email
      </div>
      <Input
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Email"
        value=""
      />
    </div>
    <div
      className="form__row"
    >
      <div
        className="form__title-input"
      >
        Логин
      </div>
      <Input
        id="login"
        name="login"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Логин"
        value=""
      />
    </div>
    <div
      className="form__row"
    >
      <div
        className="form__title-input"
      >
        Имя
      </div>
      <Input
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Имя"
        value=""
      />
    </div>
    <div
      className="form__row"
    >
      <div
        className="form__title-input"
      >
        Пароль
      </div>
      <Input
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Пароль"
        type="password"
        value=""
      />
    </div>
    <div
      className="form__row"
    >
      <div
        className="form__title-input"
      >
        Пароль (еще раз)
      </div>
      <Input
        id="passwordConfirm"
        name="passwordConfirm"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Пароль (еще раз)"
        type="password"
        value=""
      />
    </div>
    <div
      className="form__row-button"
    >
      <Memo(Button)
        disabled={true}
        onClick={[Function]}
        size="s"
        type="submit"
      >
        Зарегистрироваться
      </Memo(Button)>
    </div>
    <div
      className="form__row"
    >
      <Link
        className="form__link"
        to="/authorization"
      >
        Есть аккаунт?
      </Link>
    </div>
  </form>
</div>
`;
